### DISCUSSION TITLE
connect container' interface directly to host interface

### DISCUSSION THREAD

**Discussion by hellt** [srl-labs/containerlab#733] [Ideas]
explore if its possible to connect node' interface to host interface via the following chaining

```
[node](eth1)<---veth--->(veth_node_eth1)[host]<----tc redirect----->(ens3)[host]
```

this will allow to connect containers to the host interfaces directly.

**Comment by hmntsharma** [comment]
If my understanding is right then containerlab uses docker networking and docker host networking might just do that, simply associates the docker container's interface with host's interface, after that there is no need for publishing any port.

Going by that logic, a cRPD or FRR running as a container on a linux server, with host networking, becomes a full fledged router.
I will test this.

Please point out if I misunderstood anything here.

Cheers!

**Comment by nanisyako** [comment]
have you tried that? did it work?
